src/types/types.ts

```diff
@@ -0,0 +1,163 @@
+/**
+ * Represents options for uploading, including a function to handle previewing.
+ */
+export interface UploadOptions  { 
+  onPreview: (src: string) => void 
+};
+
+/** 
+ * User configuration of Image block tunes. Allows to add custom tunes through the config
+*/
+export interface ActionConfig { 
+  /**
+   * The name of the tune.
+   */
+  name: string;
+  
+  /**
+   * The icon for the tune. Should be an SVG string.
+   */
+  icon: string;
+  
+  /**
+   * The title of the tune. This will be displayed in the UI.
+   */
+  title: string;
+  
+  /**
+   * An optional flag indicating whether the tune is a toggle (true) or not (false).
+   */
+  toggle?: boolean;
+  
+  /**
+   * An optional action function to be executed when the tune is activated.
+   */
+  action?: Function;
+};
+
+/**
+ * UploadResponseFormat interface representing the response format expected from the backend on file uploading.
+ */
+export interface UploadResponseFormat {
+  /**
+   * success - 1 for successful uploading, 0 for failure 
+   */
+  success: number;
+
+  /**
+   * Object with file data.
+   *             'url' is required,
+   *             also can contain any additional data that will be saved and passed back
+   */
+  file: {
+    /**
+     * The URL of the uploaded image.
+     */
+    url: string;
+  };
+}
+
+/**
+ * ImageToolData type representing the input and output data format for the image tool, including optional custome actions.
+ */
+export type ImageToolData<Actions = {}> = {
+  /**
+   * Caption for the image.
+   */
+  caption: string;
+
+  /**
+   * Flag indicating whether the image has a border.
+   */
+  withBorder: boolean;
+
+  /**
+   * Flag indicating whether the image has a background.
+   */
+  withBackground: boolean;
+
+  /**
+   * Flag indicating whether the image is stretched.
+   */
+  stretched: boolean;
+
+  /**
+   * Object containing the URL of the image file.
+   */
+  file: {
```

Comment:
and here as well

Reply:
maybe we should use generic instead of `Record<string, any>` to allows users providing their types